### PR TITLE
Add Info about the OAuth2 Security Schemes to be used

### PR DESCRIPTION
**Pull request by tfrauenstein**
Improved description of security schemas and application in Zalando. Also fixed a change log entry added in #632 with incorrect rule reference.

### CODE REVIEW

File: chapters/security.adoc [hunk 1]
```diff
@@ -5,11 +5,20 @@
...
+https://swagger.io/docs/specification/authentication/[Authentication section] of the 
 Open API specification on how to specify security definitions in your API.
 
-The following code snippet shows how to define the authorization scheme using
+Most of our services require authentication via a token being passed with the 
+`Authorization: Bearer <token>` header, and here you _should_ use the 
+https://swagger.io/docs/specification/authentication/bearer-authentication/[bearerAuth security scheme] 
+for HTTP authentication (originally created as part of OAuth 2.0). 
+In situations where the service supports specific OAuth2 flows 
+-- see https://tools.ietf.org/html/rfc6750[RFC 6750], like `implicit` with redirection to authorization server --
```
Comment (tkrop) [inline chapters/security.adoc:17]: lets use `(see https://tools.ietf.org/html/rfc6750[RFC 6750])` here. I addition, you should consider to add
```
:RFC-6750: https://tools.ietf.org/html/rfc6750
```
in https://github.com/zalando/restful-api-guidelines/blob/master/index.adoc. And utilize this via `{RFC-6750}[RFC 6750]`.

**Review by tkrop — COMMENTED** [review]
While this is actually a small change it motivated me to go over the whole security chapter and to improve formatting, while remove longish explanation and redundancies. If you think this is my suggestions make sense, please merge #647 into your branch to review them together.

**Comment by tfrauenstein** [comment]
@tkrop merged your PR, thx.

**Comment by tfrauenstein** [comment]
👍

**Comment by tkrop** [comment]
If you did, you forgot to push or overrode the changes.

**Comment by tkrop** [comment]
:+1:

**Comment by vadeg** [comment]
👍

**Pull request merged**